### PR TITLE
Replace non-standard u_int64_t by t8_linearidx_t

### PR DESCRIPTION
**Pull request by sloede**
**_Describe your changes here:_**
In some locations (to be exact: 4), the non-standard type `u_int64_t` was used, which prevents compiling t8code on some platforms (e.g., Linux with musl instead of glibc). This PR fixes this by using the standardized type `uint64_t`.

cc @jmark

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] New Datatypes are added to t8indent_custom_datatypes.txt
- [x] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

### CODE REVIEW

**Review by Davknapp — CHANGES_REQUESTED** [review]
Thanks for catching that error! In all cases the variables describe a linear Id, for which we use the datatype ```t8_linearidx_t```, which is typedefed ```uint64_t```. With this change it should work.

File: src/t8_schemes/t8_default/t8_default_prism/t8_dprism_bits.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ int
 t8_dprism_compare (const t8_dprism_t *p1, const t8_dprism_t *p2)
 {
   int                 maxlvl;
-  u_int64_t           id1, id2;
+  t8_linearidx_t            id1, id2;
```
Comment (sloede) [inline src/t8_schemes/t8_default/t8_default_prism/t8_dprism_bits.c:56]:
```suggestion
  t8_linearidx_t      id1, id2;
```
Fix indentation

File: src/t8_schemes/t8_default/t8_default_tet/t8_default_tet.c [hunk 2]
```diff
@@ -59,7 +59,7 @@ static int
 t8_default_tet_compare (const t8_element_t *elem1, const t8_element_t *elem2)
 {
   int                 maxlvl;
-  u_int64_t           id1, id2;
+  t8_linearidx_t            id1, id2;
```
Comment (sloede) [inline src/t8_schemes/t8_default/t8_default_tet/t8_default_tet.c:62]:
```suggestion
  t8_linearidx_t      id1, id2;
```
Fix indentation

**Review by Davknapp — APPROVED** [review]
Thank you for your addition to t8code

**Comment by sloede** [comment]
It would be great if you can also tag v1.1.1 such that it can be used for T8code_jll.jl.

**Pull request merged**